src/models/onnxruntime_api.h

```diff
@@ -218,15 +225,31 @@ inline void InitApi() {
   //     any libonnxruntime.so that supports one of those versions.
   //
 
-  const std::string path = "libonnxruntime.so";  // "libonnxruntime4j_jni.so" is also an option if we have issues
-  void* ort_lib_handle = LoadDynamicLibraryIfExists(path);
+  void* ort_lib_handle = nullptr;
+  const char* ort_lib_path = std::getenv("ORT_LIB_PATH");
+  if (ort_lib_path) {
+    ort_lib_handle = LoadDynamicLibraryIfExists(ort_lib_path);
+  }
 
-#if !defined(__ANDROID__)
+#if defined(__linux__)
   if (ort_lib_handle == nullptr) {
+    // For Android and NuGet Linux package, the file name is libonnxruntime.so
+    // "libonnxruntime4j_jni.so" is also an option on Android if we have issues
+    ort_lib_handle = LoadDynamicLibraryIfExists("libonnxruntime.so");
+  }
+
+  if (ort_lib_handle == nullptr) {
+    // On Linux it can also be `libonnxruntime.so.1`. See: https://github.com/microsoft/onnxruntime/pull/21339
     ort_lib_handle = LoadDynamicLibraryIfExists("libonnxruntime.so.1");
   }
 #endif
 
+#if defined(MACOS_USE_DLOPEN)
+  if (ort_lib_handle == nullptr) {
+    ort_lib_handle = LoadDynamicLibraryIfExists("libonnxruntime.dylib");
```

Comment:
This is for NuGet package on macOS,. The dylib name in ORT nuget is always `libonnxruntime.dylib`